File: 2022/1/day1.py
import datetime

import common

common.init()

path = "2022/1/input.txt"


def solve() -> int:
    start = datetime.datetime.utcnow()

    data = common.load_input(path)
    part_one = part_one(data)
    part_two = part_two(data)

    runtime = datetime.datetime.utcnow() - start

    return common.Solution(1, part_one, part_two, runtime)


def part_one(data: str) -> int:
    data = [
        [int(cals) for cals in elf.split("\n") if cals != ""]
        for elf in data.split("\n\n")
    ]
    sums = [sum(items) for items in data]
    sums.sort(reverse=True)
    return sums[0]


def part_two(data: str) -> int:
    data = [
        [int(cals) for cals in elf.split("\n") if cals != ""]
        for elf in data.read().split("\n\n")
    ]
    sums = [sum(items) for items in data]
    sums.sort(reverse=True)
    return sum(sums[:3])

File: go/2022/day1/day1.go
package day1

import (
	"log"
	"sort"
	"strconv"
	"strings"
	"time"

	"github.com/jimenezj8/advent_of_code/common"
)

const input string = "2022/day1/input.txt"

func Run(root string) {
	start := time.Now()
	inputData := common.ReadInput(strings.Join([]string{root, input}, "/"))
	part1 := part1(inputData)
	part2 := part2(inputData)
	runtime := time.Now().Sub(start)

	solution := common.Solution{Day: 1, PartOne: part1, PartTwo: part2, RunTime: runtime}
	common.DisplaySolution(solution)
}

func part1(input string) (result int) {
	splitData := strings.Split(input, "\n\n")
	var elfPacks = *new([]int)

	for _, elfPack := range splitData {
		items := strings.Split(elfPack, "\n")
		total := 0
		for _, item := range items {
			if item == "" {
				continue
			}
			val, err := strconv.Atoi(item)
			if err != nil {
				log.Fatal(err)
			}
			total += val
		}
		elfPacks = append(elfPacks, total)
	}
	sort.Slice(elfPacks, func(i, j int) bool {
		return elfPacks[i] > elfPacks[j]
	})

	result = elfPacks[0]

	return result
}

func part2(input string) (result int) {
	splitData := strings.Split(input, "\n\n")
	var elfPacks = *new([]int)

	for _, elfPack := range splitData {
		items := strings.Split(elfPack, "\n")
		total := 0
		for _, item := range items {
			if item == "" {
				continue
			}
			val, err := strconv.Atoi(item)
			if err != nil {
				log.Fatal(err)
			}
			total += val
		}
		elfPacks = append(elfPacks, total)
	}
	sort.Slice(elfPacks, func(i, j int) bool {
		return elfPacks[i] > elfPacks[j]
	})

	result = 0
	for _, calories := range elfPacks[0:3] {
		result += calories
	}
	return result
}
